python/perf-kernels/flash-attention.py

```diff
@@ -617,10 +612,29 @@ def _attn_bwd_preprocess(
     else:
         tl.store(delta_ptrs, delta)
 
+@triton.jit
```

Comment:
Could you move this to the top so we have all the utility functions together? We want to move these at some point into their own file but from a dev perspective, it is better to read code if these are all above.